src/Microsoft.FeatureManagement/FeatureFilters/Cron/CronField.cs

```diff
@@ -0,0 +1,241 @@
+        private readonly int _minValue = -1;
+        private readonly int _maxValue = -1;
+        private readonly BitArray _bits;
+
+        /// <summary>
+        /// Set the min value and max value according to the field kind.
+        /// Initialize the BitArray.
+        /// </summary>
+        public CronField(CronFieldKind kind)
+        {
+            _kind = kind;
+
+            (_minValue, _maxValue) = kind switch
+            {
+                CronFieldKind.Minute => (0, 59),
+                CronFieldKind.Hour => (0, 23),
+                CronFieldKind.DayOfMonth => (1, 31),
+                CronFieldKind.Month => (1, 12),
+                CronFieldKind.DayOfWeek => (0, 7),
+                _ => throw new ArgumentException("Invalid crontab field kind.")
+            };
+
+            _bits = new BitArray(_maxValue - _minValue + 1);
+        }
+
+        /// <summary>
+        /// Checks whether the field matches the give value.
+        /// </summary>
+        /// <param name="value">The value to match.</param>
+        /// <returns>True if the value is matched, otherwise false.</returns>
+        public bool Match(int value)
+        {
+            if (!IsValueValid(value))
+            {
+                return false;
+            }
+            else
+            {
+                if (MatchAll)
+                {
+                    return true;
+                }
+
+                if (_kind == CronFieldKind.DayOfWeek && value == 0) // Corner case for Sunday: both 0 and 7 can be interpreted to Sunday
+                {
+                    return _bits[0] | _bits[7];
+                }
+
+                return _bits[ValueToIndex(value)];
+            }
+        }
+
+        /// <summary>
+        /// Checks whether the given content can be fit into the CronField.
+        /// </summary>
+        /// <param name="kind">The CronField kind. <see cref="CronFieldKind"/></param>
+        /// <param name="content">The content to parse.</param>
+        /// <param name="result">The parsed result.</param>
+        /// <returns>True if the content is valid and parsed successfully, otherwise false.</returns>
+        public static bool TryParse(CronFieldKind kind, string content, out CronField result)
+        {
+            result = null;
+            CronField cronField = new CronField(kind);
+
+            //
+            // The field can be a list which is a set of numbers or ranges separated by commas.
+            // Ranges are two numbers/names separated with a hyphen or an asterisk which represents all possible values in the field.
+            // Step values can be used in conjunction with ranges after a slash.
+            string[] segments = content.Split(',');
+            foreach (string segment in segments)
+            {
+                if (segment == null)
+                {
+                    return false;
+                }
+
+                int value = cronField.TryGetNumber(segment);
```

Comment:
I'm not a fan of calling private method on object. Can we avoid that?

Reply:
Yes. I can pass the field kind as an additional parameter and make TryGetNumber to be a static private method.